### PR TITLE
Suggested change: only show option price if its additional

### PR DESCRIPTION
**Pull request by ryanmitchell**
Only show option price where its extra to the base price to avoid end-user confusion.

### CODE REVIEW

**Comment by sampoyigi** [comment]
I like the idea but at the same time, there are cases where you want to show its free to end-users to get them excited or just for transparency so maybe adding a component property to decide whether to display 0.00 prices or not and what to display as Free or --.

Or we can leave the decision to the developer/designer, that way they can override the partial and have it display however they want.

**Comment by ryanmitchell** [comment]
Tried to add this as a component option (hideFreeOptionPrices), which is a yes/no switch, but its always coming back false in the partials even though i can see it updating in the cart.php in theme... i'm referencing it using $hideFreeOptionPrices, and its set in prepareVars() - is there another step?

Also getting an error when editing the component:
`Call to undefined method Igniter\Cart\Components\CartBox::getThemePageOptions()" on line 473
`
I've updated to latest pages, local and cart plugins to be sure its not out of date code on my side.

**Comment by sampoyigi** [comment]
Yup, that should work can't say why it's returning false, maybe if I see the code? 

That's a bug, the `getThemePageOptions ` method is defined in the `UsesPage` traits, so the CartBox class should implement it. I'll fix that :)

**Comment by ryanmitchell** [comment]
Worked it out - had to change the component in multiple pages and layouts for it to take effect everywhere.

See updates here for what I suggest doing - can be extended if the need arises, but for now its a simple on/off.

**Comment by ryanmitchell** [comment]
Sorry ,should mention I added:

`        $this->page['hideFreeOptionPrices'] = $this->property('hideFreeOptionPrices');        
`
to prepareVars() and

```
            'hideFreeOptionPrices' => [
                'label' => 'Whether to hide zero prices on options',
                'type' => 'switch',
                'default' => FALSE,
            ],
```

to defineProperties in Cartbox.php - I dont want to push that change as it will affect other PRs.

**Comment by sampoyigi** [comment]
I've merged the other PR so you can go ahead and push the changes. Also, add the hideFreeOptionPrices property effect to the option price displayed within the cartbox.

**Comment by ryanmitchell** [comment]
I've opened the PR here: https://github.com/tastyigniter/ti-ext-cart/pull/6

Not sure what you meant by: "add the hideFreeOptionPrices property effect to the option price displayed within the cartbox." - that should be done in the commit here:

https://github.com/tastyigniter/ti-theme-orange/pull/4/commits/b20bdb2a5611f77dccaf109f91cc9446ce2cb2d8

?

**Comment by sampoyigi** [comment]
Not to worry, its already done here https://github.com/tastyigniter/ti-theme-orange/blob/b20bdb2a5611f77dccaf109f91cc9446ce2cb2d8/_partials/cartBox/cart_item_options.php#L6-L8

File: _partials/cartBox/item_option_checkbox.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ class="custom-control-label w-100"
             for="menuOptionCheck<?= $optionIndex; ?>"
         >
             <?= $optionValue->name; ?>
-            <span class="pull-right"><?= currency_format($optionValue->price); ?></span>
+            <?php if ($optionValue->price > 0 || $hideFreeOptionPrices == false) {?><span class="pull-right"><?= currency_format($optionValue->price); ?></span><?php } ?>
```
Comment (sampoyigi) [inline _partials/cartBox/item_option_checkbox.php:23]: Also for readability, change the if statements to `$optionValue->price > 0 || !$hideZeroOptionPrices`. It's also missing the + sign.

Comment (ryanmitchell) [reply _partials/cartBox/item_option_checkbox.php]: Done :)

**Pull request merged**